pandas/_libs/tslibs/offsets.pyx

```diff
@@ -233,7 +233,7 @@ def _to_dt64(dt, dtype='datetime64'):
     # numpy.datetime64('2013-05-01T02:00:00.000000+0200')
     # Thus astype is needed to cast datetime to datetime64[D]
     if getattr(dt, 'tzinfo', None) is not None:
-        i8 = pydt_to_i8(dt)
+        i8 = dt.timestamp() * 10**9
```

Comment:
Is there any chance of losing precision doing it this way? I think can be somewhere around the 2 ** 60 range for floats

Reply:
I don't think so, pydt obects only have 6 digits after the decimal, so we should be OK.  Besides, later in this function we case to datetime64[D]